includes/Utilities/Tracker.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package FacebookCommerce
+ */
+
+namespace SkyVerge\WooCommerce\Facebook\Utilities;
+
+defined( 'ABSPATH' ) or exit;
+
+/**
+ * Class for adding diagnostic info to WooCommerce Tracker snapshot.
+ *
+ * See https://woocommerce.com/usage-tracking/ for more information.
+ *
+ * @since %VERSION%
+ */
+class Tracker {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since %VERSION%
+	 */
+	public function __construct() {
+		add_filter(
+			'woocommerce_tracker_data',
+			array( $this, 'add_tracker_data' )
+		);
+	}
+
+	/**
+	 * Append our tracker properties.
+	 *
+	 * @param array $data The current tracker snapshot data.
+	 * @return array $data Snapshot updated with our data.
+	 * @since %VERSION%
+	 */
+	public function add_tracker_data( array $data = array() ) {
+		$connection_handler = facebook_for_woocommerce()->get_connection_handler();
+		if ( ! $connection_handler ) {
+			return $data;
+		}
+
+		if ( ! isset( $data['extensions'] ) ) {
+			$data['extensions'] = array();
+		}
+
+		$connection_is_happy = $connection_handler->is_connected() && ! get_transient( 'wc_facebook_connection_invalid' );
+		$data['extensions']['facebook']['connected'] = $connection_is_happy;
```

Comment:
Instead of `facebook`, I think the plugin's directory name (`facebook-for-woocommerce`) is better because it is more distinctive. This could avoid naming collisions with other Facebook-related plugins in the future.

Reply:
I was thinking the same thing. I think I had an irrational fear of `-` in keys, which you have cured 😄

Reply:
Now back to `extensions_facebook-for-woocommerce_is-connected` which is more explicit and more consistent (with plugin slug).